image-pipeline/expression-scatter-plots.js

```diff
@@ -223,24 +236,25 @@ async function configureIntercepts(page) {
       const headers = Object.assign({}, request.headers())
       const [isESPlot, gene] = detectExpressionScatterPlot(request)
       if (isESPlot) {
+        // TODO: Retain this block for now
         // Replace SCP API request for expression data with prefetched data.
-        //
-        // If these files could be made by Ingest Pipeline and put in a bucket,
-        // then Image Pipeline could run against production web app while
-        // incurring virtually no load for app server or DB server, and likely
-        // complete warming a study's image cache 5-10x faster.
         const extension = values['json-dir'] ? '.gz' : ''
-        const jsonGzPath = `${jsonFpStem + gene }.json${extension}`
-        const content = await readFile(jsonGzPath)
+        // const jsonGzPath = `${jsonFpStem + gene }.json${extension}`
+        // const content = await readFile(jsonGzPath)
+
+        const content = expressionByGene[gene]
 
-        let jsonString
         if (extension === '.gz') {
-          const arrayString = strFromU8(gunzipSync(content))
-          initExpressionResponse.data.expression = JSON.parse(arrayString)
-          jsonString = JSON.stringify(initExpressionResponse)
-        } else {
-          jsonString = content
+          // TODO: Retain this comment block for now.
+          // Gzipping data _in scatter data pipeline_ prior bucket upload
```

Comment:
Interesting - so we would write `gzip` data to the bucket but not set the `Content-Encoding` header?  That makes sense provided that `strFromU8(gunzipSync(content))` doesn't incur any performance hits.  We'd likely want to append `.gz` to the files then to denote that they are gzipped.